### DISCUSSION TITLE
Non Euclidean doors (linked_portal_door)

### DISCUSSION THREAD

**Discussion by xLeo340x** [BEEmod/BEE2-items#4097] [Feature Suggestions]
Some maps created in Hammer have non-Euclidean spaces created using transparent "portal walls" that the player can't distinguish from normal corridors during the game. I think you have seen these many times, it would be great to see such in Bee2

**Comment by slambo311** [comment]
Just saying, most people just call them world portals. But yeah, world portals would be great in BEE2. There's already world portals connecting the elevator rooms to the main test chamber, so why not have some in the actual test?

**Comment by BenVlodgi** [comment]
We considered it, they would be tricky to design, and are error prone in game. If you have more than one in an area, you can expect terrible results.
I do like using them in hammer, some of my earliest maps used these to achieve a gravity changing field barrier effect.
https://www.youtube.com/watch?v=6wQrLisdeEk
https://www.youtube.com/watch?v=z2fMmzMRiQg

**Comment by vrad-exe** [comment]
Actually I found a way to get rid of the 4 portals onscreen limit, although there are still some additional things to consider for having world portals in Puzzlemaker:
- How would placing them work? Ideally they should be resizeable like glass, but glass can be split up by walls or otherwise made into irregular shapes, while world portals always have to be rectangular.
- Lighting doesn't transfer through portals, and matching up lighting across the portal can already be hard in Hammer, let alone in Puzzlemaker where you don't have manual control over light entities.
- Portals are one of the most expensive things in the game to render, which isn't something most Puzzlemaker mappers will be aware of.
- There are a few other bugs with world portals, for example if you have a laser enter a world portal without a wall directly behind it, it'll be offset when coming out the other end.

**Reply by slambo311** [reply]
Well, you could add some warnings in the description mentioning these, similar to how the reflection gel's description tells you not to change the gel type.

**Reply by vrad-exe** [reply]
The difference is that the reflection gel crash is just a bug in the editor, whereas this would affect published maps as well.

**Reply by slambo311** [reply]
Alright, but still, just telling mappers about it would work. And for the irregular shape stuff, making it a fizzler could help.

**Reply by vrad-exe** [reply]
How would making it a fizzler solve any of these problems?

**Reply by slambo311** [reply]
Well, placed like a fizzler. Fizzlers can't really be made into irregular shapes like glass, so it'd help that first problem.

**Reply by vrad-exe** [reply]
Fizzlers can be made into irregular shapes too.
